ee/server/service/calendar.go

```diff
@@ -63,6 +71,60 @@ func (svc *Service) CalendarWebhook(ctx context.Context, eventUUID string, chann
 		return authz.ForbiddenWithInternal(fmt.Sprintf("calendar channel ID mismatch: %s != %s", savedChannelID, channelID), nil, nil, nil)
 	}
 
+	lockValue, reserved, err := svc.getCalendarLock(ctx, eventUUID, true)
+	if err != nil {
+		return err
+	}
+	// If lock has been reserved by cron, we will need to re-process this event in case the calendar event was changed after the cron job read it.
+	if lockValue == "" && !reserved {
+		// We did not get a lock, so there is nothing to do here
+		return nil
```

Comment:
This is when our server could not get a lock (because another server has the lock). In the `svc.getCalendarLock` method, we added this event to the queue. So, the other server will re-process this event.

I could add a debug log here, but this might happen often in production.